### ISSUE TITLE
DisableErrorsInGeneratedCode

### ISSUE THREAD

**Issue by vorburger** [google/error-prone#808]
Would you consider adding a `DisableErrorsInGeneratedCode` option?

As its name implies, it would, just like it's existing inspiration `DisableWarningsInGeneratedCode`, prevent errors from being raised on [anything annotated with @Generated](https://github.com/google/error-prone/pull/807).

I'm expecting you :innocent:  to reply and close this as you _"NEVER!",_ because _"why would anyone possibly want to ignore errors even in generated code?!",_ but here's the rationale:

Some of error-prone's (great!) checks are not _necessarily_ flagging clear errors only. For example, [`Var` is more of a best practice style which can help prevent errors - but a violation of it is not an error.](http://errorprone.info/bugpattern/Var) I really like `Var` though, and have cleaned up an existing code base to that style (because I now have one small reason less to adopt Kotlin and Xtend which have this same concept), and would now like to enforce it to fail builds running on Gerrit code review if my colleagures re-introduce unnecessary ugly `final`.

But I cannot use e.g. `-Xep:Var:ERROR` because my project uses a code generator which is not (yet) peperring the @Var annotation over it's generated code - so I'm stuck. Of course one can try to get that code generator to comply - but, generally speaking, that is not realistic - nor neccesary.

**Comment by vorburger** [comment]
In this specific case, [I've suggested that the Immutables.org code generator could emit `@Var`](https://github.com/immutables/immutables/issues/716), but there will be other people running into similar isuses with other Bug Patterns and other Code Generators.

**Comment by msridhar** [comment]
FYI now that #599 has landed you can probably achieve this by excluding the paths to the generated code

**Comment by vorburger** [comment]
@msridhar yeah as soon as #599 is actually released... waiting for https://github.com/google/error-prone/issues/821

**Comment by vorburger** [comment]
[I now use XepExcludedPaths to achieve what I originally filed this issue for](https://git.opendaylight.org/gerrit/#/c/66590/); just using like:

`-XepExcludedPaths:.*/target/.*`

So at least from my side this issue could just be closed now, there's no "need" for this, actually.

**Comment by xenoterracide** [comment]
I mean, I'm using -XepExlcludedPaths too, but I also kind of like the idea of ignoring things based on annotation. perhaps even just excluding generated code by default because this is a tool to audit my team's code not generated, and not libraries.

**Comment by davido** [comment]
This can be closed now?
